pkg/config/expand.go

```diff
@@ -103,8 +103,7 @@ func (dc *DeploymentConfig) addMetadataToModules() error {
 	for iGrp, grp := range dc.Config.DeploymentGroups {
 		for iMod, mod := range grp.Modules {
 			if mod.RequiredApis == nil {
-				_, projectIDExists := dc.Config.Vars["project_id"].(string)
-				if !projectIDExists {
+				if dc.Config.Vars.Get("project_id").Type() != cty.String {
```

Comment:
When I wrote this code, I inserted this validation logic here because (going by memory), one could choose to construct RequiredAPIs in such a way that one didn't need to define `project_id`. I think it may make more sense to  change the logic slightly:

1. in `validateVars`, enforce logic similar to `vars.labels`. if `project_id` exists, it must be a `cty.String`
2. At this point in the code, switch to `! Has("project_id")`.

Reply:
I would like to keep this PR free of behavior changes.